### PR TITLE
Fix CategoricalArrays.jl in Project.toml

### PR DESCRIPTION
**Pull request by bkamins**
@KristofferC - I have a question regarding Project.toml.

If a package is in `[extras]` then is `[compat]` entry allowed for it?

In https://julialang.github.io/Pkg.jl/v1/toml-files/#The-[compat]-section it is stated that only `[deps]` entries are allowed in `[compat]`, but maybe also `[extras]` are allowed and the description given there should be fixed? Thank you!

### CODE REVIEW

**Comment by simeonschaub** [comment]
> If a package is in [extras] then is [compat] entry allowed for it?

Yes, that should work.

**Comment by bkamins** [comment]
So I understand the documentation in Pkg.jl should be fixed - right?

**Comment by simeonschaub** [comment]
Yes. Could you open an issue?

**Comment by bkamins** [comment]
https://github.com/JuliaLang/Pkg.jl/issues/2424

File: Project.toml [hunk 1]
```diff
@@ -24,7 +23,7 @@ Tables = "bd369af6-aec1-5ad0-b16a-f7cc5008161c"
 Unicode = "4ec0a83e-493e-50e2-b9ac-8f72acf5a8f5"
 
 [compat]
-CategoricalArrays = "0.8.3, 0.9"
```
Comment (nalimilan) [inline Project.toml:27]: Do we really need 0.9?

Comment (bkamins) [reply Project.toml:27]: we need 0.9.3 for `unwrap`

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you

**Pull request merged**